### PR TITLE
fix #1630, preserve indentation of package.json files and default to 2 spaces, similar to NPM

### LINKED ISSUE #1630: bit init always indent package.json to 4 spaces, should respect original formatting
## Expected Behavior
After bit init, the diff should be:
```diff
diff --git a/package.json b/package.json
index c3c4c05..6223055 100644
--- a/package.json
+++ b/package.json
@@ -5,5 +5,10 @@
   "license": "MIT",
   "devDependencies": {
     "bit-bin": "^14.1.0"
+  },
+  "bit": {
+    "env": {},
+    "componentsDefaultDirectory": "components/{name}",
+    "packageManager": "npm"
   }
 }

```

## Actual Behavior
But diff is:
```diff
diff --git a/package.json b/package.json
index c3c4c05..434da9d 100644
--- a/package.json
+++ b/package.json
@@ -1,9 +1,14 @@
 {
-  "name": "myrepro",
-  "version": "1.0.0",
-  "main": "index.js",
-  "license": "MIT",
-  "devDependencies": {
-    "bit-bin": "^14.1.0"
-  }
-}
+    "name": "myrepro",
+    "version": "1.0.0",
+    "main": "index.js",
+    "license": "MIT",
+    "devDependencies": {
+        "bit-bin": "^14.1.0"
+    },
+    "bit": {
+        "env": {},
+        "componentsDefaultDirectory": "components/{name}",
+        "packageManager": "npm"
+    }
+}
\ No newline at end of file
```

## Steps to Reproduce the Problem
run:
```bash
mkdir myrepro && cd myrepro && yarn init -y && yarn add -D bit-bin && git init && git add package.json yarn.lock && git commit -m "initial commit" && yarn bit init  && git diff
```

## Specifications

  - Bit version: 14.1.0
  - Node version: v8.16.0
  - npm / yarn version: 1.15.2
  - Platform: Mac
  - Bit compiler (include version): ?
  - Bit tester (include version): ?

### PR DESCRIPTION
**Pull request by davidfirst**
This change is [Reviewable](https://reviewable.io/reviews/teambit/bit/1636)

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -341,6 +341,8 @@ export const WRAPPER_DIR = 'bit_wrapper_dir';
 
 export const PACKAGE_JSON = 'package.json';
 
+export const PACKAGE_JSON_INDENT = 2; // as per NPM default
```
Comment (GiladShoham) [inline src/constants.js]: PACKAGE_JSON_INDENT -> PACKAGE_JSON_DEFAULT_INDENT

File: src/api/consumer/lib/doctor.js [hunk 2]
```diff
@@ -128,7 +128,7 @@ async function _generateExamineResultsTarFile(examineResults: ExamineResult[]):
     pack.entry({ name: '.bitmap' }, bitmap);
   }
   if (consumerInfo && consumerInfo.consumerConfig) {
-    pack.entry({ name: 'config.json' }, consumerInfo.consumerConfig.toJson());
+    pack.entry({ name: 'config.json' }, JSON.stringify(consumerInfo.consumerConfig.toPlainObject(), null, 4));
```
Comment (GiladShoham) [inline src/api/consumer/lib/doctor.js]: Why did you use here 4 and not 2 from the constants?

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [14.1.1-dev.1] - 2019-05-14
 
+- preserve indentation of `package.json` files and default to 2 spaces, similar to NPM (#1630)
```
Comment (GiladShoham) [inline CHANGELOG.md]: move to unreleased

File: src/api/consumer/lib/doctor.js (hunk 2 above)
Comment (davidfirst) [reply src/api/consumer/lib/doctor.js]: coz it's for the doctor, to be compliant with other json files there, which are 4. let me know if you want to change.

File: src/constants.js (hunk 1 above)
Comment (davidfirst) [reply src/constants.js]: yeah, I was debating between the two. ok, I'll go with the default.

File: src/api/consumer/lib/doctor.js (hunk 2 above)
Comment (GiladShoham) [reply src/api/consumer/lib/doctor.js]: anyway we should put it in constants

**Pull request merged**